x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -67,6 +69,16 @@ export interface ProviderSession {
    * Cookie "Path" attribute that is validated against the current Kibana server configuration.
    */
   path: string;
+
+  /**
+   * The set of flags used to describe various aspects of the user session.
+   */
+  flags?: {
```

Comment:
__note:__ if we wouldn't have plans to switch to server side sessions, I'd use a bitmask field here instead to save space :slightly_smiling_face:

Reply:
That's actually a good point, I had something else in mind at the beginning, but now I can't think of anything. Maybe `flags` is too premature indeed, I'll switch to the top level property for now, thanks!